src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -266,6 +240,37 @@ namespace winrt::TerminalApp::implementation
         CATCH_LOG();
     }
 
+    winrt::fire_and_forget TerminalPage::NewTerminalByDrop(winrt::Windows::UI::Xaml::DragEventArgs& e)
+    {
+        Windows::Foundation::Collections::IVectorView<Windows::Storage::IStorageItem> items;
+        try
+        {
+            items = co_await e.DataView().GetStorageItemsAsync();
+        }
+        CATCH_LOG();
+
+        if (items.Size() == 1)
+        {
+            std::filesystem::path path(items.GetAt(0).Path().c_str());
+            if (!std::filesystem::is_directory(path))
+            {
+                path = path.parent_path();
+            }
+
+            std::wstring pathText = path.wstring();
+
+            // Handle edge case of "C:\\", seems like the "StartingDirectory" doesn't like path which ends with '\'
```

Comment:
Wait what? My experience with `StartingDirectory` tells me this can't be true ...

Reply:
This edge case happens when you press `shift` and drop the `C:` drive on it, you get the following error:

> [error 0x8007010b when launching `cmd.exe']
> Could not access starting directory "C:\WINDOWS\SysWOW64\""

D: drive
>[error 0x8007010b when launching `cmd.exe']
>Could not access starting directory "D:""

While trying to debug the value of `pathText` I notice normal folders don't end with `\` while `C` & `D` drives do.
removing that `\` solved the problem...maybe it solved it by avoiding it?
can you please help me understand why it happens?
note that pressing `shift` is critical since normal drop works just fine.
If you think it should work without this `if` will trying to debug it further.